runtime/lua/vim/lsp/handlers.lua

```diff
@@ -341,17 +341,21 @@ function M.signature_help(_, method, result, _, bufnr, config)
     print('No signature help available')
     return
   end
-  local lines = util.convert_signature_help_to_markdown_lines(result)
-  lines = util.trim_empty_lines(lines)
-  if vim.tbl_isempty(lines) then
-    print('No signature help available')
-    return
-  end
-  local syntax = api.nvim_buf_get_option(bufnr, 'syntax')
-  local p_bufnr, _ = util.focusable_preview(method, function()
-    return lines, util.try_trim_markdown_code_blocks(lines), config
+  local p_bufnr, winnr = util.focusable_float(method, function()
+    local ft = api.nvim_buf_get_option(bufnr, 'filetype')
+    local lines = util.convert_signature_help_to_markdown_lines(result, ft)
+    lines = util.trim_empty_lines(lines)
+    if vim.tbl_isempty(lines) then
+      print('No signature help available')
+      return
+    end
+    local p_bufnr, p_winnr = util.fancy_floating_markdown(lines, {
+      border = config.border
```

Comment:
`hover` only allows `border` to be set, so I took this over, while the original method passed the full config.
Maybe I should change this so that both `hover` and `signature_help` pass the full config?

Reply:
If you do that, maybe you can also adapt https://github.com/neovim/neovim/blob/75992f2ad9189a5f39b3e2ae7962c68d3af815a4/runtime/lua/vim/lsp/util.lua#L946-L966 while you're at it? 😁 (If it's not too much work.)

Reply:
(Ah, keep forgetting that this uses a different code path 😞 )

Reply:
@clason the bug there is that they should pass `filetype` instead of `syntax` I guess?
I assume when you use that function, you don't get proper highlighting in the popup?

Treesitter disables syntax, so ft probably would fix it.

Reply:
Just checked it should definitely be `filetype` to get highlighting.

Apart from that, wouldn't it make more sense to simply load the `targetUri` buffer in the popup with the definition on top? Then people can focus in the popup if they want and even make edits.

Reply:
Sorry, I should have been cleaner. I didn't mean the bug (which I personally never experienced, but it makes sense to fix it) but the passing of border options.

The rest of the function behaves as intended -- it should _not_ show the full buffer but only the `LocationRange`; this might not be useful for some servers, but it definitely is for the one I'm using (texlab). 

But I can see that what you propose would be useful, but then it should be a separate function (`make_focusable_preview`) -- and a different PR so this can be discussed properly.

Reply:
Maybe it makes sense to ignore my comment (and this function) for now.

Reply:
@clason totally makes sense. I'll keep the ft / syntax fix in here, and just added the ability to pass options to the popup window, but won't add anything else related to `preview_location`